analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -106,6 +109,31 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
                 return type.IsAny(KnownType.NonIntegralNumbers);
             }
 
+            if (division.Ancestors().LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x)) != null)
+            {
+                if (division.Ancestors().OfType<AssignmentExpressionSyntax>().FirstOrDefault() is { } assignmentSyntax
+                    && assignmentSyntax.MapAssignmentArguments() is { } assignmentMappings)
+                {
+                    var assignementLeft = assignmentMappings.Where(x => x.Right.Equals(division)).FirstOrDefault().Left;
+                    if (assignementLeft != null)
+                    {
+                        type = semanticModel.GetTypeInfo(assignementLeft).Type;
+                        return type.IsAny(KnownType.NonIntegralNumbers);
+                    }
+                }
+                else if (division.Ancestors().OfType<VariableDeclarationSyntax>().FirstOrDefault() is { } variableDeclaration)
+                {
+                    var tupleArguments = ((TupleExpressionSyntaxWrapper)division.Parent.Parent).AllArguments();
+                    var declarationType = semanticModel.GetTypeInfo(variableDeclaration.Type).Type;
+                    var tupleTypes = (declarationType as INamedTypeSymbol)?.TupleElements().Select(x => x.Type).ToArray();
+                    if (tupleTypes != null)
+                    {
+                        type = tupleTypes[tupleArguments.IndexOf((ArgumentSyntax)division.Parent)];
+                        return type.IsAny(KnownType.NonIntegralNumbers);
+                    }
```

Comment:
I think this is the better approach here than MapAssignmentArguments:
* This is working on the semanticModel and covers more use cases.
* The `(INamedTypeSymbol)?.TupleElements()` should be available available for all 4 sides:
  * Declaration:
    *  variableDeclaration.Type
    * variableDeclaration.Initializer
  * Assignment
    * (assignment,Left as DeclarationExpression).Designation (either via IOperation or one of the GetSymbolInfo, GetDeclaredSymbol methods)
    * (assignment,Left as TupleExpression)

If the above works, you end up with two ITypeSymbol for both sides and a tuple element symbol representing the division. What you need next is the "path" of the division in the tuple. Because of the nesting, this is a list of indexes from top to bottom:

`var (a, (b, c)) = (1, (1/3, 2)); // Indizes:[1, 0]`

You need to walk up from `1/3` and store for the indexes (e.g. in a stack). And then you need to walk down on the other side along the indexes to find `b`. `b` is some kind of symbol and you can call `symbol.GetSymbolType().IsAny(KnownType.NonIntegralNumbers)` and via `ISymbol.GetDeclaredSyntaxReferences` you may find the definition in code.

This is all theory by me and I don't know for sure if it is working. But if it does, it should come in quite handy. A helper function could be extracted:
`ISymbol FindCorrespondingSymbol(ISymbol origin, ISymbol nestedInOrigin, ISymbol destination)`
 We need to discuss offline, whether we want to give it a try or not.

Reply:
I added the two examples you mentioned above and I also changed the way that the division and tuples are discovered so now they are supported.

It you be indeed nice to create something more generic in the helpers.

Reply:
I also added  `(int, (decimal, (int, decimal))) nested = (3, (1 / 3, (0, 1 / 3)));` this UT and a method that flattens a tuple (but it's a tuple of Types). Not sure if it worth adding it to the helpers!